src/components/interpolatedColorLegend.ts

```diff
@@ -0,0 +1,249 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Component {
+  export class InterpolatedColorLegend extends AbstractComponent {
+    private _measurer: SVGTypewriter.Measurers.Measurer;
+    private _wrapper: SVGTypewriter.Wrappers.Wrapper;
+    private _writer: SVGTypewriter.Writers.Writer;
+    private _scale: Scale.InterpolatedColor;
+    private _orientation: String ;
+    private _padding = 5;
+    private _formatter: Formatter;
+
+    private _swatchContainer: D3.Selection;
```

Comment:
A swatch is a color sample. The term is reasonably common (in paint and color selection) and describes the purpose of the `<rect>`s that are created.

http://en.wiktionary.org/wiki/swatch